File: backend/src/main/java/de/neuefische/readandmeet/backend/model/Meeting.java
package de.neuefische.readandmeet.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@Document(collection = "meetings")
public class Meeting {

    @Id
    private final String id;
    private String title;
    private LocalDate date;
    private String location;
    private Book book;
}

File: frontend/src/models/meeting.ts
export type Meeting = {
    id: string,
    title: string,
    date: string,
    location: string,
    book?: {
        id?: string;
        title?: string;
        author?: string;
    }
    userId: string;
}


export type MeetingWithoutId = {
    title: string,
    date: string,
    location: string,
    book?: {
        id?: string;
        title?: string;
        author?: string;
    }
}
